commcare_connect/opportunity/forms.py

```
@@ -1096,106 +1097,152 @@ def __init__(self, *args, **kwargs):
         invoice_number = self.cleaned_data["invoice_number"]
-        if PaymentInvoice.objects.filter(opportunity=self.opportunity, invoice_number=invoice_number).exists():
+
+        if not invoice_number:
+            invoice_number = self.generate_invoice_number()
+
+        if PaymentInvoice.objects.filter(invoice_number=invoice_number).exists():
             raise ValidationError(
-                f'Invoice "{invoice_number}" already exists',
+                "Please use a different invoice number",
                 code="invoice_number_reused",
             )
```

Ruff (0.14.5)

Avoid specifying long messages outside the exception class

(TRY003)